### PR TITLE
added ClippingRegionNode

### PR DESCRIPTION
**Pull request by SunLightJuly**
ClippingRegionNode is Useful supplement of ClippingNode, more efficient and compatible

### CODE REVIEW

File: cocos/2d/CCClippingRegionNode.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+class CC_DLL ClippingRegionNode : public Node
+{    
+public:
+    static ClippingRegionNode* create(const Rect& clippingRegion);
+    static ClippingRegionNode* create(void);
+    
+    const Rect getClippingRegion(void) {
+        return _clippingRegion;
+    }
```
Comment (minggo) [inline cocos/2d/CCClippingRegionNode.h]: return `const Rect&`

Comment (minggo) [inline cocos/2d/CCClippingRegionNode.h]: const Rect getClippingRegion(void) -> const Rect& getClippingRegion(void) const?

File: cocos/2d/CCClippingRegionNode.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    void setClippingRegion(const Rect& clippingRegion);
+    
+    bool isClippingEnabled(void) {
+        return _clippingEnabled;
+    }
```
Comment (minggo) [inline cocos/2d/CCClippingRegionNode.h]: Please add const

File: cocos/2d/CCClippingRegionNode.h (hunk 1 above)
Comment (SunLightJuly) [inline cocos/2d/CCClippingRegionNode.h]: modified

File: cocos/2d/CCClippingRegionNode.h (hunk 2 above)
Comment (SunLightJuly) [inline cocos/2d/CCClippingRegionNode.h]: fixed

**Comment by minggo** [comment]
@dabingnn 
Could you please review it?

**Comment by dabingnn** [comment]
@SunLightJuly 
Thanks, I found that ClippingRegionNode is implemented by Scissor Test, which is used for screen aligned clipping. Is this node capable of clipping a rotated sprite?

**Comment by SunLightJuly** [comment]
ClippingRegionNode is more effectively,  but it is able to clip a rect only

**Comment by SunLightJuly** [comment]
See new PR:   https://github.com/cocos2d/cocos2d-x/pull/8706

**Pull request closed without merging**